### PR TITLE
pythonPackages.jsbeautifier: init at 1.6.4.0

### PR DESCRIPTION
**Pull request by apeyroux**
###### Motivation for this change

This little beautifier will reformat and reindent bookmarklets, ugly JavaScript, unpack scripts packed by Dean Edward’s popular packer, as well as deobfuscate scripts processed by javascriptobfuscator.com.

https://github.com/beautify-web/js-beautify
###### Things done
- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/python-modules/jsbeautifier/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{ stdenv, fetchurl, python, wrapPython, pytest }:
+
+stdenv.mkDerivation rec {
```
Comment (FRidh) [inline pkgs/development/python-modules/jsbeautifier/default.nix]: this is a python package. Please use `buildPythonPackage` and put the whole expression in `python-packages.nix`. Check the expressions in that file to see how it should be used.

**Review by FRidh — CHANGES_REQUESTED** [review]

**Comment by 41px** [comment]
I had to add editorconfig.

**Comment by FRidh** [comment]
you might want to go through the contributing guidelines as well

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -1407,7 +1407,7 @@ in modules // {
       maintainers = with maintainers; [ abbradar ];
     };
   };
-
+
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: whitespace

File: pkgs/top-level/python-packages.nix [hunk 3]
```diff
@@ -6182,7 +6182,22 @@ in modules // {
     };
   });
 
+  editorconfig = buildPythonPackage rec {
+    name = "editorconfig";
+
+    src = pkgs.fetchgit {
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: PyPI is the preferred source

File: pkgs/top-level/python-packages.nix (hunk 2 above)
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: whitespace

File: pkgs/top-level/python-packages.nix [hunk 4]
```diff
@@ -6182,7 +6182,22 @@ in modules // {
...
+      url = "https://github.com/editorconfig/editorconfig-core-py.git";
+      rev = "582fbd04f9b31eb518afda56e2fc5378d24a07d6";
+      sha256 = "0gapdrnw6wc75vq867ki2vhsynqn82jb80knf1d704db99cgbnjb";
+    };
 
+    meta = {
+      homepage = "http://editorconfig.org";
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: urls are special and do not need quotations

File: pkgs/top-level/python-packages.nix [hunk 5]
```diff
@@ -7317,6 +7332,23 @@ in modules // {
     };
   };
 
+  jsbeautifier = buildPythonApplication rec {
+    name = "jsbeautifier-1.6.4";
+
+    buildInputs = with self; [ editorconfig pytest six ];
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: Are you sure it needs six only during building, and not during run-time?
Also, did you check whether the tests are actually run with all interpreter versions?

**Review by FRidh — CHANGES_REQUESTED** [review]

**Review by kevincox — APPROVED** [review]

**Comment by FRidh** [comment]
As I pointed out before, please go through the contributing guidelines. All changes need to be in logical units (commits) and named according to the guidelines.

**Comment by 41px** [comment]
I can delete it and recreate it with the contributing guidelines.

Le 26 sept. 2016 10:05 PM, "Frederik Rietdijk" <notifications@github.com> a
écrit :

**Comment by FRidh** [comment]
ping

**Comment by apeyroux** [comment]
see #19163

**Pull request closed without merging**